flink-runtime/src/main/java/org/apache/flink/runtime/deployment/InputGateDeploymentDescriptor.java

```diff
@@ -54,24 +56,31 @@
 	 * The index of the consumed subpartition of each consumed partition. This index depends on the
 	 * {@link DistributionPattern} and the subtask indices of the producing and consuming task.
 	 */
+	@Nonnegative
 	private final int consumedSubpartitionIndex;
 
 	/** An input channel for each consumed subpartition. */
-	private final InputChannelDeploymentDescriptor[] inputChannels;
+	private final ShuffleDescriptor[] inputChannels;
+
+	/**
+	 * {@link ResourceID} of partition consume to identify its location.
+	 *
+	 * <p>It can be used e.g. to compare with partition producer {@link ResourceID} in
+	 * {@link ProducerDescriptor} to determine producer/consumer co-location.
+	 */
+	private final ResourceID consumerLocation;
```

Comment:
consumerLocation -> consumerResourceId?  Because in `ProducerDescriptor` or `NettyShuffleDescriptor`, we also name `producerResourceId`

Reply:
actually I think it is better to rename it in `ProducerDescriptor & NettyShuffleDescriptor `